Helper/AdyenOrderPayment.php

```diff
@@ -214,7 +214,7 @@ public function createAdyenOrderPayment(Order $order, Notification $notification
     {
         $adyenOrderPayment = null;
         $payment = $order->getPayment();
-        $amount = $this->adyenDataHelper->originalAmount($notification->getAmountValue(), $order->getBaseCurrencyCode());
+        $amount = $this->adyenDataHelper->originalAmount($notification->getAmountValue(), $order->getOrderCurrencyCode());
```

Comment:
Hi @acampos1916 , 

thanks for replying back. 

Not sure if the question is directed to me but I took the liberty to respond: 
Looking at the plugin code, I believe both options will work as expected. 

However, I would stick to the Magento default $order->getOrderCurrencyCode() as that's following the formatting that Magento needs always. 

If you use $notification->getAmountCurrency() and in the future, you decide to change the code how webhooks pass the currency code or decide not to pass it at all, it will break the function again. 

In the end, it's a matter of preference :). 

I applied my solution on our live website and so far we didn't experience any bugs or problems after applying the patch. 

Best
Dejena

Reply:
Thanks @dejankar. If I get it right there are multiple scenarios to cover:

Base currency A, Display currency B (getOrderCurrencyCode != getAmountCurrency)
Base currency B, Display currency A (getOrderCurrencyCode != getAmountCurrency) 
Base currency A, Display currency A (getOrderCurrencyCode == getAmountCurrency)
Base currency B, Display currency B (getOrderCurrencyCode == getAmountCurrency)

If the charged currency config included in the plugin is set to Display the payment will be processed with the Display currency, but the getOrderCurrencyCode would return the Base currency. I think it is safer to use `$notification->getAmountCurrency()` because of that, what do you think?

Reply:
Hey @dejankar, of course! Thanks for helping us figure this out.

I'll test your patience a bit more 😄 since the notification is being processed after the payment has been completed there's a small chance that the plugin's config for charged currency has been modified between the order placement and the webhook being received, that would set the wrong currency with your updated approach.

I would indeed suggest to use the `$notification->getAmountCurrency()`, to my knowledge there are no plans at all to change the format from the Adyen side and hasn't been changed for a while. It is safe to use.

Reply:
Hi @acampos1916 , 

Sure it's my pleasure. 

Yeah, I agree there is a small chance for the mentioned use case but in such config, the whole process might be wrong. 
There might be a problem if let's say the order was placed in base currency and then someone changes the charged currency setting and then webhooks returns the values with a different currency. 

That might interfere with Magento's internal logic for let's say if someone invoice later and want to capture the amount in charged currency. Than anyway the order needs to be fixed from the M2 side. 

But all in all, I'm really fine with the proposal from your side: $notification->getAmountCurrency(), especially as you mentioned that format won't be changed in the future, and the field will always be called like this. 

I will leave the decision for you and your team! 

It's been a pleasure chatting with all of you guys :)

Looking forward to the solution and all the great stuff that this plugin offer.

Reply:
@dejankar good point! We do have a mechanism in place so that orders are captured and refunded in the currency with which it was placed, see the docblock for [this helper function](https://github.com/Adyen/adyen-magento2/blob/develop/Helper/ChargedCurrency.php#L52-L80).

I'm doing a suggestion with the use of the notification currency. Thank you!